### ISSUE TITLE
Improve source strings handling and visibility

### ISSUE THREAD

**Issue by nijel** [WeblateOrg/weblate#2831]
**Is your feature request related to a problem? Please describe.**
Currently source strings are tracked by separate model what leads to several problems:

* UI wise the source strings and their checks are hidden somewhere in the navigation, it should be at same level as translations to get higher visibility
* Editing the priority or flags anyway leads to updating all other units, so the source really does not matter here
* The UI for source strings is completely different to translations and is ugly. This is to biggest extend caused by it's invisibility - nobody cares.

**Describe the solution you'd like**

* Track source strings as regular translations. This is already there for monolingual files with editable base, for rest it needs to be implemented.
* Adjust translation and translate UI to better accommodate source strings view (show source checks, source comments, make the string non editable if bilingual or configured so, probably depends on #743).
* Adjust comment notification to link to source string for source string comments (should address #2812)

**Steps needed**

* [x] Add support for non translatable strings, https://github.com/WeblateOrg/weblate/issues/743
* [x] Move Source model logic to Unit on source translation
* [x] Migrate Source metadata to Unit, creating source language Translation
* [x] Change the `edit_template` logic to add read-only flag on translation
* [x] Integrate Source view into current translate views (context, screenshots, ...)
* [x] Rethink `source.edit` and `unit.template` permissions

**Comment by comradekingu** [comment]
I really like this, clicking into specific projects per language to get every instance of every check gets old quickly.

**Comment by nijel** [comment]
Thank you for your report, the issue you have reported has just been fixed. 

* In case you see a problem with the fix, please comment on this issue. 
* In case you see similar problem, please open separate issue.
* If you are happy with the outcome, consider supporting Weblate by [donating](https://weblate.org/donate/).
